sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitStmt.cs

```diff
@@ -454,6 +454,15 @@ private void VisitDoStmt(DoStmt doStmt)
 
         private void VisitExplicitCastExpr(ExplicitCastExpr explicitCastExpr)
         {
+            if (IsPrevContextDecl<EnumConstantDecl>(out var _) && explicitCastExpr.Type is EnumType)
```

Comment:
I don't think removing the cast is going to be correct in all cases
imagine a scenario where the value is `-1` and the target enum base type is `long` or `uint`

It's likely "safer" to insert an additional cast to the base type instead or possibly to replace it with a cast to the target enum base type

Reply:
Okay, it will insert a cast to the underlying type instead now.